### PR TITLE
startup 6 #1-2

### PR DESCRIPTION
**Pull request by Adrenno**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,7 +27,7 @@ gem 'sequel'
 gem 'sqlite3'
 gem 'hirb'
 gem 'logger'
-gem 'ostruct'
```
Comment (Arijus1979) [inline Gemfile:30]: ostruct gem is needed

File: app/controllers/app.rb [hunk 2]
```diff
@@ -1,55 +1,176 @@
-# frozen_string_literal: true
```
Comment (Arijus1979) [inline app/controllers/app.rb]: This line is needed in every file

File: app/controllers/app.rb [hunk 3]
```diff
@@ -1,55 +1,176 @@
...
+require 'figaro'
 require 'airports'
+require 'securerandom'
 require_relative '../infrastructure/amadeus/gateways/amadeus_api'
 require_relative '../infrastructure/nytimes/gateways/nytimes_api'
 require_relative '../infrastructure/database/repositories/flights'
+require 'logger'  # Add this line to require Logger
 
 module WanderWise
-  # Main application class for WanderWise
```
Comment (Arijus1979) [inline app/controllers/app.rb:11]: comments are needed for rubocop

File: app/infrastructure/amadeus/gateways/amadeus_api.rb [hunk 4]
```diff
@@ -8,24 +8,19 @@
 module WanderWise
   # Gateway to Amadeus API for flight offers data
   class AmadeusAPI
+    class AmadeusAPIError < StandardError; end
     def initialize
       environment = ENV['RACK_ENV'] || 'development'
       
-      # Only load secrets from secrets.yml in development/test environments
       if environment == 'development' || environment == 'test'
-        secrets_file_path = './config/secrets.yml'
```
Comment (Arijus1979) [inline app/infrastructure/amadeus/gateways/amadeus_api.rb:16]: this logic is needed for production environment, keep it like it was

File: app/infrastructure/amadeus/gateways/amadeus_api.rb [hunk 5]
```diff
@@ -8,24 +8,19 @@
...
-          raise "secrets.yml file not found for #{environment} environment."
-        end
+        secrets = YAML.load_file('./config/secrets.yml')
+        @client_id = secrets[environment]['amadeus_client_id']
+        @client_secret = secrets[environment]['amadeus_client_secret']
       else
-        # For production, use environment variables
-        @client_id = ENV['amadeus_client_id']
-        @client_secret = ENV['amadeus_client_secret']
+        @client_id = ENV['AMADEUS_CLIENT_ID']
```
Comment (Arijus1979) [inline app/infrastructure/amadeus/gateways/amadeus_api.rb]: lowercase 'amadeus_client_id' and 'amadeus_client_secret'

File: app/infrastructure/nytimes/gateways/nytimes_api.rb [hunk 6]
```diff
@@ -9,25 +9,17 @@
 module WanderWise
   # Gateway to NY Times API for recent articles
   class NYTimesAPI
-    def initialize
```
Comment (Arijus1979) [inline app/infrastructure/nytimes/gateways/nytimes_api.rb:12]: this logic is needed for production environment, keep previous changes

File: app/infrastructure/nytimes/gateways/nytimes_api.rb [hunk 7]
```diff
@@ -9,25 +9,17 @@
...
 
+    def initialize
+      environment = ENV['RACK_ENV']
+      secrets = YAML.load_file('./config/secrets.yml')
+      @secrets = secrets[environment]
+      @api_key = @secrets['nytimes_api_key']
       @base_url = 'https://api.nytimes.com/svc/search/v2/articlesearch.json'
 
-      # Create a fixture file for the API response if it doesn't exist in development/test
-      save_to_fixtures unless File.exist?('./spec/fixtures/nytimes-api-results.yml') || environment == 'production'
```
Comment (Arijus1979) [inline app/infrastructure/nytimes/gateways/nytimes_api.rb:30]: same, needed for production

File: app/infrastructure/nytimes/gateways/nytimes_api.rb [hunk 8]
```diff
@@ -55,27 +46,17 @@ def save_to_fixtures
...
       response_body = response.body.to_s.force_encoding('UTF-8')
       JSON.parse(response_body)
-    rescue HTTP::Error => e
-      handle_error(e)
-    end
-
-    private
-
-    # Load secrets from secrets.yml for development/test environments
-    def load_secrets
```
Comment (Arijus1979) [inline app/infrastructure/nytimes/gateways/nytimes_api.rb:68]: load secrets is needed in the initialize method

File: config/environment.rb [hunk 9]
```diff
@@ -4,7 +4,6 @@
 require 'yaml'
 require 'figaro'
 require 'sequel'
-require 'ostruct'
```
Comment (Arijus1979) [inline config/environment.rb]: ostruct is needed here

File: ponse to a fixture file
Comment (Arijus1979) [inline ponse to a fixture file:1]: what is this file?

**Review by Arijus1979 — CHANGES_REQUESTED** [review]

**Review by Adrenno — COMMENTED** [review]
all resolved. pls check

File: ponse to a fixture file
Comment (Adrenno) [reply ponse to a fixture file:1]: it was likely accidentally created while i was working on the tests. basically it's an instruction for the "less" command in wsl for viewing text file contents. we can just delete it

File: app/infrastructure/amadeus/gateways/amadeus_api.rb [hunk 10]
```diff
@@ -21,30 +22,50 @@ def initialize
         else
           raise "secrets.yml file not found for #{environment} environment."
         end
+        secrets = YAML.load_file('./config/secrets.yml')
```
Comment (Arijus1979) [inline app/infrastructure/amadeus/gateways/amadeus_api.rb]: these 3 lines are not needed here I think

File: app/infrastructure/nytimes/gateways/nytimes_api.rb [hunk 11]
```diff
@@ -23,7 +23,13 @@ def initialize
         secrets = load_secrets
         @api_key = secrets[environment]['nytimes_api_key']
       end
+    class Error < StandardError; end
 
+    def initialize
```
Comment (Arijus1979) [inline app/infrastructure/nytimes/gateways/nytimes_api.rb]: is this needed? the "initialize" function is already defined above

**Review by Arijus1979 — CHANGES_REQUESTED** [review]

File: ponse to a fixture file
Comment (Arijus1979) [reply ponse to a fixture file:1]: delete it before pushing it then

**Pull request merged**